crates/miden-objects/src/asset/vault/partial.rs

```diff
@@ -190,7 +191,7 @@ mod tests {
             assert_eq!(entry, invalid_asset);
         });
 
-        let err = PartialVault::default().add(proof).unwrap_err();
+        let err = PartialVault::default().add(AssetWitness::new_unchecked(proof)).unwrap_err();
```

Comment:
```suggestion
        let err = PartialVault::default().add(AssetWitness::new(proof)?).unwrap_err();
```

Let's use `Asset::new` so if this ever gets out of sync, we don't get a corrupted asset witness.